### ISSUE TITLE
Lastest 4chan update broke the viewer

### ISSUE THREAD

**Issue by ghost** [wakarimasenco/ChanExplorer#11]
> Title

**Comment by mksvco** [comment]
still waiting for a fix :-/

**Comment by BenAbdul** [comment]
Chan explorer is my favourite 4chan browser but right now is unusable. Please respond

**Comment by ghost** [comment]
This happened a lingua ti me ago, but it was eventually fixed... no it broke again... also why si ChanExplorer remo ed from the Sto re?

**Comment by mksvco** [comment]
@miyachan ?

**Comment by ghost** [comment]
Please @Miyachan Chan Explorer is the Best 4chan browser imho, now I'm stuck with Chanu (garbage)

**Comment by BenAbdul** [comment]
A moment of silence for Chan Explorer

**Comment by miyachan** [comment]
Sorry guys, I've been super busy with other things.

The fix isn't too hard and source is there, but I'll update in the coming weeks sorry guys.

**Comment by ghost** [comment]
@miyachan Yes, you're alive \* w \* Thank you very much, I'll wait for the fix ^^

**Comment by slooblack** [comment]
HI,
The size and the title on the image viewer are broke. I think 4chan change the HTML code so the parser fail.

@miyachan  i try to fix this problem, I fix the post.size, but i'm still suck on the post.name. I don't find the picture name on the HTML code.

**Comment by BoBeR182** [comment]
Did the latest commit #12 fix it? Does anyone have a working APK?

@miyachan can you update f-droid repos?

**Comment by miyachan** [comment]
Sorry for abandoning this project guys. Can someone post a screenshot so I can see exactly whats broken?

**Comment by BoBeR182** [comment]
It just says "There was an error loading this page" on every board you attempt to open and will have the spinning wheel of eternal loading.

@miyachan 
![screenshot_2014-07-17-04-36-42](https://cloud.githubusercontent.com/assets/525433/3610569/0b68b816-0d8e-11e4-8e1f-dc3ae89a6484.png)

**Comment by D4NH** [comment]
![2014-07-14 09 52 59](https://cloud.githubusercontent.com/assets/6664547/3569303/4e3d85b6-0b3d-11e4-892f-254e39377e05.png)

**Comment by mksvco** [comment]
4 months later.. :/

**Comment by ghost** [comment]
I really hope there are some serious problema to keep him/her from answering theese comments, or it's just not fair, this id the best 4Chan Browser for Android imho, I mo ed to Chanu and it's buggy as hell. Hope you answer soon, :)

**Comment by mksvco** [comment]
I've tried forking it and checking it out but the code is an enormous mess.

**Comment by miyachan** [comment]
Long story short, the revenue this app gave me was the only incentive I had to work on it. I'm completely broke nowadays and I told myself I would work on it when my situation changes, but it hasn't.

The issue is with the parser, ChanExplorer has been in development longer than the API has been around, so it works by scraping the HTML. There are two options
1. Update the parser to work with the new html: https://github.com/wakarimasenco/ChanExplorer/blob/master/src/co/wakarimasen/chanexplorer/imageboard/Parser.java
   
   It essentially works by identifying what the interesting blocks of text are surrounded by and pulling those. It could be as small as some html has changed, I'm not completely sure, but you would have to look through it.
2. Rework parse to work with the 4chan api. This takes considerably more work , but it essentially involves changing (https://github.com/wakarimasenco/ChanExplorer/blob/master/src/co/wakarimasen/chanexplorer/ThreadFragment.java#L841) to do a getrequest on the API url, and parsing the resulting JSON to return a `[]Post`

The first way would be the easiest but it would require sitting down with a text editor to capture the relevant areas. The second way would require less code and be forever compatible to html changes but may require refactoring (especially for users that don't have a 4chan pass)

**Comment by mksvco** [comment]
Thanks for the update @miyachan. I'm going to try to get it working.

**Comment by mksvco** [comment]
Well, I'm getting this error trying to compile. https://gist.github.com/MikeSavi/ee7014252384d070d316 . Would you know what this is ?

**Comment by miyachan** [comment]
What is in the file : C:\Users\mike\Desktop\ChanExplorer\app\build\intermediates\ndk\debug\Android.mk

Whatever in this file should be the same as
https://github.com/wakarimasenco/ChanExplorer/blob/master/jni/Android.mk

**Comment by davidlohle** [comment]
I'll take over code development. A rewrite for the API might happen in the future, but for now I'm just going to update the HTML parser to get everything working.

Follow development here: https://github.com/Proplex/Leaf

**Comment by davidlohle** [comment]
![device-2014-08-16-153002](https://cloud.githubusercontent.com/assets/2215601/3942870/c51cce00-257b-11e4-9331-7d8e973d4f35.png)

Got it working; just fixing up the post formatting.

**Comment by D4NH** [comment]
looking good! keep it up!

**Comment by davidlohle** [comment]
![device-2014-08-17-021204](https://cloud.githubusercontent.com/assets/2215601/3943904/c9e183a4-25d5-11e4-9d76-c83000a71cae.png)
Giving the app a much needed facelift, as it existed prior to the xxhdpi screens, etc. Expect a Google Play-ready app by the next few days.

https://github.com/Proplex/Leaf

**Comment by miyachan** [comment]
@Proplex I don't know what the criteria is but ChanExplorer was my 3rd 4Chan app removed off the playstore (4Chon, ChanExplorer (with Trollface icon), ChanExplorer (with /4/ icon))

I think Mimi and Chanu have survived, and ChanExplorer is much older than those (I made ChanExplorer as a competitor to ChanDroid - which no longer exists either) and I'm not sure how they have done it.

**Comment by davidlohle** [comment]
@miyachan For what reasons? And for reference, I used all three versions of your app (all other apps are straight-up ass), as well as ChanDroid.

I'll poke around and see what's up. Did you mark the app as high maturity?

For those wanting the revived ChanExplorer: ( @MikeSavi @demonabis )
https://play.google.com/store/apps/details?id=co.wakarimasen.ceredux

**Comment by miyachan** [comment]
All the same reason (with High Maturity):

```
Violation of the sexually explicit material provision of the Content Policy.
```

This was with all the NSFW boards hidden (you had to go to into the settings to enable them). FWIW this only really happened when ChanExplorer was the most popular 4chan app (it hovered around ~#60 in top apps in Social), and it may be the case of someone reporting the app.

However Mimi and Chanu are still around.

In any case thanks for taking over development - I didn't really mean to abandon ChanExplorer, life pretty much got in the way, and I'm glad to see it active again.

**Comment by davidlohle** [comment]
Glad I could help. I'll be replacing the HTML parser pretty soon, the API isn't hard at all and it'll really be smooth that way. Have a few ideas for a tablet redesign too.

**Comment by ghost** [comment]
Nice to see ChanExplorer reborn. I hope Leaf will find it's way to F-Droid as well once it's ready.

**Comment by davidlohle** [comment]
@chiffon-rouge It definitely will. I'm in the process of moving into my dorm these next two days and the source code/development environment is on my desktop which is packed away. Once I get setup I'll setup a F-Droid repo.

**Comment by ghost** [comment]
What's the status on either CE or Leaf? Right now, unfortunately, both projects seem to be inactive.
